scripts/spike.js

```diff
@@ -0,0 +1,228 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+const faker = require('faker')
+const { merge } = require('lodash')
+const crypto = require('crypto')
+const fetch = require('node-fetch')
+const { version } = require('../packages/rum/package.json')
+
+/**
+ * To make the random id generation work.
+ */
+const destination = new Uint8Array(16)
+global.crypto = {
+  getRandomValues: () => {
+    return crypto.randomFillSync(destination)
+  }
+}
+const {
+  generateRandomId
+} = require('../packages/rum-core/dist/lib/common/utils')
+
+const defaultMeta = {
+  metadata: {
+    service: {
+      name: 'benchmark-spike',
+      agent: {
+        name: 'rum-js',
+        version
+      },
+      language: {
+        name: 'javascript'
+      }
+    }
+  }
+}
+const defaultTransaction = {
+  transaction: {
+    id: '73a0d4714e793b8a',
+    trace_id: '2b282fc14bd5a45bd69798c0c70fbe53',
+    name: 'test-transaction',
+    type: 'page-load',
+    duration: 3092,
+    context: {
+      page: {
+        referer: '',
+        url: 'https://www.elastic.co/'
+      },
+      response: {
+        transfer_size: 333,
+        encoded_body_size: 28019,
+        decoded_body_size: 140745
+      }
+    },
+    marks: {
+      navigationTiming: {
+        fetchStart: 0,
+        domainLookupStart: 0,
+        domainLookupEnd: 0,
+        connectStart: 0,
+        connectEnd: 0,
+        requestStart: 50,
+        responseStart: 157,
+        responseEnd: 157,
+        domLoading: 161,
+        domInteractive: 540,
+        domContentLoadedEventStart: 607,
+        domContentLoadedEventEnd: 610,
+        domComplete: 1350,
+        loadEventStart: 1350,
+        loadEventEnd: 1350
+      },
+      agent: {
+        timeToFirstByte: 157,
+        domInteractive: 540,
+        domComplete: 1350
+      }
+    },
+    span_count: {
+      started: 82
+    },
+    sampled: true,
+    experience: {
+      tbt: 100,
+      cls: 0.5,
+      fid: 50
+    }
```

Comment:
include breakdown metrics as well for transactions and spans?

Reply:
Added the transaction breakdown, the span break downs currently would add one more document since we currently have one type in this script